bootenv_test.go

```diff
@@ -128,3 +132,38 @@ func Test_EnvRead_NonExisting_FailsReading(t *testing.T) {
 		t.FailNow()
 	}
 }
+
+func Test_EnvCanary(t *testing.T) {
+	runner := newTestOSCalls("var=1\nmender_check_saveenv_canary=1\nmender_saveenv_canary=0\n", 0)
+	fakeEnv := uBootEnv{&runner}
```

Comment:
Why not make this a table test?

Reply:
I looked, but I think it's not worth it for this small test. There are only 6 test cases, and 3 of them take different code paths.